src/pages/sign-up/SignUp.tsx

```diff
@@ -1,5 +1,195 @@
-function SignUp() {
-  return <h1>SignUp</h1>
+import React, { useState } from 'react'
+import { Formik, Form, Field, ErrorMessage } from 'formik'
+import * as Yup from 'yup'
+import { AiFillEyeInvisible, AiFillEye } from 'react-icons/ai'
+import { useNavigate } from 'react-router-dom'
+interface MyFormValues {
+  email: string
+  fname: string
+  lname: string
+  phoneNumber: string
+  password: string
+}
+const SignUp: React.FC<MyFormValues> = () => {
+  const navigate = useNavigate()
+  const [showPassword, hidePasword] = useState(false)
+  console.log(showPassword)
+  const triggerEyes = () => {
+    hidePasword(!showPassword)
+  }
+  const initialValues: MyFormValues = {
+    email: '',
+    fname: '',
+    lname: '',
+    phoneNumber: '',
+    password: '',
+  }
+  return (
+    <div className="max-w-lg p-10 m-4 mx-auto">
+      <Formik
+        initialValues={initialValues}
+        onSubmit={(values, actions) => {
+          console.log(values)
+          actions.setSubmitting(false)
+          navigate('/signin')
+        }}
+        validationSchema={Yup.object().shape({
+          email: Yup.string()
+            .email('Invalid email')
+            .strict(false)
+            .trim()
+            .required('*Email is required'),
+          fname: Yup.string()
+            .max(15, 'Too Long!')
+            .strict(false)
+            .trim()
+            .required('*Firstname is required'),
+          lname: Yup.string()
+            .max(15, 'Too Long!')
+            .strict(false)
+            .trim()
+            .required('*Lastname is required'),
+          phoneNumber: Yup.number()
+            .typeError('Enter valid Phone Number')
+            .positive("A phone number can't start with a minus")
+            .integer("A phone number can't include a decimal point")
+            .min(12, 'Enter valid Phone Number'),
+          password: Yup.string().required('*Password is required'),
+        })}
+      >
+        {({ errors, touched }) => (
+          <Form className="bg-white rounded px-2 pt-2 pb-2 mb-4 max-w-xl">
+            <div className="mb-6">
+              <h1 className="bg text-3xl font-semibold content-start">
+                Create an account
+              </h1>
+            </div>
+            <div className="mb-6">
+              <label
+                className="block text-black text-sm font-semibold mb-2"
```

Comment:
'block text-black text-sm font-semibold mb-2' * 5 times in the same file can be given as a combined CSS class using @apply annotation